src/components/Sing/ExportOverlay.vue

```diff
@@ -0,0 +1,72 @@
+<template>
+  <div v-if="nowExporting" class="export-overlay">
+    <div>
+      <QSpinner color="primary" size="2.5rem" />
+      <div class="q-mt-xs">
+        {{
+          nowRendering
+            ? "レンダリング中・・・"
+            : `${exportingMediaName}を書き出し中・・・`
+        }}
+      </div>
+      <QBtn
+        v-if="nowRendering"
+        padding="xs md"
+        :label="`${exportingMediaName}の書き出しをキャンセル`"
+        class="q-mt-sm"
+        outline
+        @click="cancelExport"
+      />
+    </div>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { computed } from "vue";
+import { useStore } from "@/store";
+
+const store = useStore();
+
+const nowRendering = computed(() => {
+  return store.state.nowRendering;
+});
+const nowExporting = computed(() => {
+  return store.state.exportState !== "NOT_EXPORTING";
+});
+const exportingMediaName = computed(() => {
+  if (store.state.exportState === "EXPORTING_AUDIO") {
+    return "音声";
+  } else if (store.state.exportState === "EXPORTING_LABEL") {
+    return "labファイル";
+  } else {
+    return "";
```

Comment:
あ、ここ到達しないはずなのでエラーにしてあげるほうがよさそう！
```suggestion
    throw new Error(store.state.exportState);
```
到達しないはずだと開発者側にわかるように`UnreachableError`をimportして使うとより良さそう！

```suggestion
    throw new UnreachableError(store.state.exportState);
```

Reply:
エラーをthrowするようにすると、
- `v-if="nowExporting"`されている要素の中で`exportingMediaName`が使用されている
- 不要な場合にcomputedの処理が走らない

のを前提としたコードになり（`script`タグ内のコードが`template`タグ内のコードに依存している）、その辺りで違和感があったので、refとwatchを使用する形にしました。

（`script`タグ内のコードが`template`タグ内のコードに依存しない形が良いかなと思います）

Reply:
なるほどです！！
たしかにtemplate内での位置を考えながらcomputedの中を実装するのは良くなさそう！！

ただwatchにすると結局`""`が生まれてしまうので、問題は変わってないかも･･･？
直和型を使って、`nowExporting`がtrueのときのみ`exportingMediaName`があることを型レベルで制限してしまうのがロジック的には一番正しいかもです！（一時的な状態を排除できる）

ということで、直和型（ユニオン型？）のお手本みたいな感じなので、ちょっとPR作ってみました！！
参考になれば！！

- https://github.com/sigprogramming/voicevox/pull/3

Reply:
PRありがとうございます！マージ＆少し変更させていただきました！